charmcraft/charm_builder.py

```diff
@@ -134,7 +136,10 @@ def handle_generic_paths(self):
                 rel_path = rel_basedir / name
                 abs_path = abs_basedir / name
 
-                if self.ignore_rules.match(str(rel_path), is_dir=True):
+                if name == STAGING_VENV_DIRNAME:
```

Comment:
Extended `ignore_rules` and removed special handling for this directory:
```
DEBUG    :: Translated .jujuignore 1 "/staging-venv" => "/staging\-venv\Z"
...
DEBUG    :: Building charm in '/home/claudio/charmcraft/charmcraft/testdir/build/parts/charm/install'
DEBUG    :: Linking in generic paths
DEBUG    :: Ignoring directory because of rules: 'staging-venv'
```